lib/web/scripts/oneoff/oneoff.go

```diff
@@ -130,13 +153,37 @@ func (p *OneOffScriptParams) CheckAndSetDefaults() error {
 	}
 	p.CDNBaseURL = strings.TrimRight(p.CDNBaseURL, "/")
 
-	if p.TeleportFlavor == "" {
-		p.TeleportFlavor = types.PackageNameOSS
+	if p.SupportedOSes == nil {
+		p.SupportedOSes = []string{constants.LinuxOS, constants.DarwinOS}
+	}
+
+	if p.TeleportDirectory == "" {
+		p.TeleportDirectory = types.PackageNameOSS
+		if modules.GetModules().BuildType() == modules.BuildEnterprise {
+			p.TeleportDirectory = types.PackageNameEnt
+		}
+	}
+	if p.TeleportArtifact == "" {
+		p.TeleportArtifact = types.PackageNameOSS
 		if modules.GetModules().BuildType() == modules.BuildEnterprise {
-			p.TeleportFlavor = types.PackageNameEnt
+			p.TeleportArtifact = types.PackageNameEnt
 		}
 	}
-	if !slices.Contains(types.PackageNameKinds, p.TeleportFlavor) {
+	// TODO(vapopov): DELETE IN v20.0.0, `teleport-update` must be already added to all supported
+	// releases and version check must be omitted.
+	version, err := utils.EnsureSemver(p.TeleportVersion)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Versions 17.7.2 and 18.1.5 are when the separate `teleport-update` package was introduced.
+	if version.Major == 17 && version.Compare(semver.Version{Major: 17, Minor: 7, Patch: 2}) >= 0 ||
+		version.Major == 18 && version.Compare(semver.Version{Major: 18, Minor: 1, Patch: 5}) >= 0 ||
+		version.Major > 18 {
+		p.TeleportDirectory = types.PackageNameOSS
+		p.TeleportArtifact = types.PackageNameUpdate
```

Comment:
If I'm understanding this correctly, this means we always use the `teleport-update` binary when using those versions.

Will this work for integration configurations?
In those situations, we don't want to install teleport, we just want to execute a `teleport integrations configure ....` command on the host, and that's it.

I think it's fine to fetch the `teleport-update` package to then fetch the teleport binary.
But, will we be executing the `teleport integrations configure ...` afterwards?